src/streaming/MediaPlayer.js

```diff
@@ -1288,6 +1290,79 @@ function MediaPlayer() {
         return textController.isTextEnabled();
     }
 
+    /**
+     * Use this method to add a captions from an external URL (for captions not listed in the manifest).
+     * @param {string} url - URL of the captions file to load.  Use module:MediaPlayer#dashjs.MediaPlayer.events.TEXT_TRACK_ADDED.
+     * @param {MediaInfo} mediaInfo - A MediaInfo object with details about the text track.
+     * @see {@link MediaPlayerEvents#event:TEXT_TRACK_ADDED dashjs.MediaPlayer.events.TEXT_TRACK_ADDED}
+     * @throws {@link module:MediaPlayer~PLAYBACK_NOT_INITIALIZED_ERROR PLAYBACK_NOT_INITIALIZED_ERROR} if called before initializePlayback function
+     * @memberof module:MediaPlayer
+     * @instance
+     */
+    function addTextTrack(url, mediaInfo) {
+        if (!streamingInitialized) {
+            throw STREAMING_NOT_INITIALIZED_ERROR;
+        }
+
+        if (textController === undefined) {
+            textController = TextController(context).getInstance();
+        }
+
+        /********* END ***********/
+        let captionsLoader = createCaptionsLoader();
```

Comment:
Do we need to create a new instance of CaptionsLoader every time? Might consider creating a single instance as part of TextController and then call textController.getCaptionsLoader

Reply:
I think you're right, but I'm not sure how to do it. Currently the `CaptionsLoader` inherits `requestModifier` and `mssHandler` from the `MediaPlayer` config. Those don't seem available from inside `TextController`, so I'm not sure how pass those along if it's created in `TextController.setup()`.